src/lib/translators/mongo.ts

```diff
@@ -432,7 +432,79 @@ function transformPivot(step: Readonly<S.PivotStep>): MongoStep[] {
   ];
 }
 
-/** transform an 'replace' step into corresponding mongo steps */
+/** transform a 'rank' step into corresponding mongo steps */
+function transformRank(step: Readonly<S.RankStep>): MongoStep {
+  if (step.column) {
+    return [
+      {
+        $group: {
+          _id: `$${step.column}`,
+          grouped_ranked_column: {
+            $push: '$$ROOT',
+          },
+        },
+      },
+      {
+        $sort: {
+          _id: step.sortOrder == 'asc' ? 1 : -1,
+        },
+      },
+      {
+        $group: {
+          _id: 'whatever',
+          items: {
+            $push: '$$ROOT',
+          },
+        },
+      },
+      {
+        $unwind: {
+          path: '$items',
+          includeArrayIndex: 'items.rank',
+        },
+      },
+      {
+        $unwind: {
+          path: '$items.grouped_ranked_column',
+        },
+      },
+      {
+        $replaceRoot: {
+          newRoot: {
+            $mergeObjects: [
+              '$items.grouped_ranked_column',
+              { [step.rankColumnName || 'Rank']: '$items.rank' },
+            ],
+          },
+        },
+      },
+    ];
+  } else {
```

Comment:
You do not need this. It's strange to rank something based on nothing :)
Also, as already said, it will never be true because columnPicker always returns a column, and it's good!